components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/token/handlers/grant/AbstractAuthorizationGrantHandler.java

```diff
@@ -492,14 +499,22 @@ public boolean validateGrant(OAuthTokenReqMessageContext tokReqMsgCtx)
                 throw new IdentityOAuth2Exception(e.getMessage(), e);
             }
         }
```

Comment:
Can we set the OAuthAppDO as a message context property in AccessTokenIssuer and use it here? This could potentially reduce a DB call in the worst case.